addons/im_livechat/static/tests/helpers/mock_server.js

```diff
@@ -252,4 +252,14 @@ patch(MockServer.prototype, 'im_livechat', {
             ...livechats,
         ];
     },
+    /**
+     * @override
+     */
+    _mockResUsersSettings_ResUsersSettingsFormat(id) {
+        const [settings] = this.getRecords('res.users.settings', [['id', '=', id]]);
+        return {
+            ...this._super(...arguments),
+            is_discuss_sidebar_category_livechat_open: settings.is_discuss_sidebar_category_livechat_open,
+        };
+    },
```

Comment:
there should be no override, and base code returning all fields instead

also this whole "use res.users.settings fields in JS code" commit is really badly split

makes no sense to have change in server or mock server in such commit

Reply:
> and base code returning all fields instead

Without formatting, we retrieve unwanted fields added by the MockServer (`__last_update`, `name`, `display_name`), and all the automatic fields that are normally filtered server side.

Implementing the same logic here would require to list all the fields to be removed, is that what I should do?

Reply:
Yes I prefer to list the fields to remove in this case, same as in python in the mock server.